crates/apollo-compiler/src/validation/field.rs

```diff
@@ -10,11 +11,41 @@ use crate::{
 pub fn validate_field(
     db: &dyn ValidationDatabase,
     field: Arc<hir::Field>,
+    type_def: hir::TypeDefinition,
 ) -> Vec<ApolloDiagnostic> {
     let mut diagnostics =
         db.validate_directives(field.directives().to_vec(), hir::DirectiveLocation::Field);
     diagnostics.extend(db.validate_arguments(field.arguments().to_vec()));
 
+    let field_type = field.ty(db.upcast());
+    if let Some(field_type) = field_type {
+        // Get the type system definition for the type of the field - is there a better way to do this?
+        let field_type_def = field_type.type_def(db.upcast());
+
+        if let Some(field_type_def) = field_type_def {
+            diagnostics
+                .extend(db.validate_selection_set(field.selection_set().clone(), field_type_def));
+        } else {
+            // TODO what should we do if field_type_def is None although field_type is Some? Is that a case we are expecting?
```

Comment:
this would be impossible as `field.ty()`, the function that's called here to get `field_type` runs the same function as field_type_def - `db.find_type_definition_by_name()`. so if `field_type` is Some, field_type_def is always going to be Some